tests/compiletest.rs

```diff
@@ -27,6 +27,15 @@ fn run_pass() {
     compiletest::run_tests(&config);
 }
 
+fn miri_pass(path: &str, target: &str) {
+    let mut config = compiletest::default_config();
+    config.mode = "mir-opt".parse().expect("Invalid mode");
```

Comment:
mir-opt is not ideal

1. it runs the optimization passes (but I think we run miri before those are run)
2. it generates a final binary that is then executed on the host.

The clean solution would be to add a "compile-pass" test to compiletest